runtime/compiler/runtime/RelocationRuntime.cpp

```diff
@@ -1434,3 +1426,30 @@ TR_JITaaSRelocationRuntime::allocateSpaceInDataCache(uintptr_t metaDataSize,
    return newDataStart;
    }
 
+uint8_t *
+TR_JITaaSRelocationRuntime::copyDataToCodeCache(const void *startAddress, size_t totalSize, TR_J9VMBase *fe)
+   {
+   TR::CompilationInfoPerThreadBase *compInfoPT = fe->_compInfoPT;
+   int32_t numReserved;
+   TR::CodeCache *codeCache = compInfoPT->getCompilation()->cg()->getCodeCache();
```

Comment:
So the thing is, `RelocationRuntime` doesn't have a pointer to a valid code cache until `prepareRelocateAOTCodeAndData` is called. At that point, code cache is set to `cg()->getCodeCache()`, which will be also NULL on the client, so we would allocate a new code cache there.
I decided to change the code for `copyDataToCodeCache`, so that whenever we allocate code cache there, we set `cg()->setCodeCache` so that at the end of the compilation we can use the same code cache for the rest of the code received from the server.
So with my latest change, both cache pointers should point to the same cache.

Reply:
There is a danger that, when trying to allocate code for the compiled body there isn't sufficient space because, during reservation we only requested that the size of the thunks be available. It's better to let the client reserve again a new code cache that is appropriate in size.